R/bs4_book.R

```diff
@@ -316,7 +316,7 @@ tweak_footnotes <- function(html) {
   id <- xml2::xml_attr(footnotes, "id")
   xml2::xml_remove(xml2::xml_find_all(footnotes, "//a[@class='footnote-back']"))
   contents <- vapply(footnotes, FUN.VALUE = character(1), function(x) {
-    as.character(xml2::xml_children(x))
+    paste0(as.character(xml2::xml_children(x)), collapse = "")
```

Comment:
@cderv this fixes the bug you noticed.

How should we go about adding an integration test that'd be a book with all these sorts of things? So we'd catch a regression that wouldn't allow multiline footnotes anymore for instance.

Reply:
We could create a demo book that contains any features we think off and that we render to different format. We have https://github.com/rstudio/bookdown-demo but it does not contain much in content for now as the aim is to show how to organize a simple bookdown project. 

For `gibook()` the bookdown book is built regularly on dev version and helps find building error.

We could try add to bookdown-demo or create something new. But this could only show building error as other issues (like the mathjax in footnote) would require manual visual test.

Reply:
And a manual visual test is probably more efficient than building a pipeline with automatic clicking and screenshots :sweat_smile:

Reply:
How would one store the list of things to check manually, though?

Reply:
If you want to go this road, you could just write the checklist as the content of your document I guess.

Reply:
I added an example for this in bookdown-demo (https://github.com/rstudio/bookdown-demo/pull/63). This will serve as test, and we can then book the build in several version for test. I need to do a GHA workflow for this. We'll see later.